### PR TITLE
mocha chai and sinon dependecies added

### PR DESCRIPTION
**Pull request by christianR15c**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "bcrypt": "^5.0.1",
+    "chai": "^4.3.6",
+    "debug": "^4.3.3",
+    "dotenv": "^16.0.0",
+    "express": "^4.17.2",
+    "http-errors": "^2.0.0",
+    "jsonwebtoken": "^8.5.1",
+    "mocha": "^9.2.0",
+    "morgan": "^1.10.0",
+    "sinon": "^13.0.1",
```
Comment (SimonImanigirimpuhwe) [inline package.json]: Please installing packages is not enough, consider making a full configuration to make sure testing is working

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (christianR15c) [reply package.json]: ok, let me work on it

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import chai from 'chai';
```
Comment (JPaul23) [inline test/test.js]: @christianR15c , remember to include some pictures running these tests man.

Comment (christianR15c) [reply test/test.js]: Ok, noted. I am going to add them

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: akhileshns/heroku-deploy@v3.12.12
+        with:
+          heroku_api_key: ${{secrets.b22f31da-9e25-401e-b066-b96bc29c3819}}
+          heroku_app_name: 'phantom-api-pirates'
+          heroku_email: 'habchris15@gmail.com'
+
```
Comment (NyiingangoPacifique) [inline .github/workflows/main.yml:40]: your heruko_api_key should be stored in .env file

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# This is a basic workflow to help you get started with Actions
+
```
Comment (David-Neza) [inline .github/workflows/main.yml:2]: create a proper folder structure that makes it easy to add dependencies for postgres and others.

File: .github/workflows/main.yml (hunk 3 above)
Comment (christianR15c) [reply .github/workflows/main.yml:40]: I see, I am going to add them in .env file

File: src/app.js [hunk 5]
```diff
@@ -38,7 +41,11 @@ app.use('/api-docs', swaggerUI.serve, swaggerUI.setup(swaggerDocs))
 app.use(welcomeRoute)
 
 // PORT
+<<<<<<< HEAD
 const port = config.app.port
+=======
+const port = process.env.PORT || 3000
+>>>>>>> configure CI/CD using Github Actions and Heroku
```
Comment (SimonImanigirimpuhwe) [inline src/app.js]: What is this @christianR15c ? Please make sure you rebase appropriately

File: test/test.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import chaiHTTP from 'chai-http';
```
Comment (SimonImanigirimpuhwe) [inline test/test.js]: This module is not installed in your dependencies, please run test again you will see it complaining about the module

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

File: test/test.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+const chai = require('chai');
+const chaiHTTP = require('chai-http');
+const { app } = require('../src/app');
+
```
Comment (SimonImanigirimpuhwe) [inline test/test.js:4]: Good

**Review by SimonImanigirimpuhwe — APPROVED** [review]

**Pull request merged**